common/persistence/sql/sqlHistoryManager.go

```diff
@@ -480,15 +480,14 @@ func (m *sqlHistoryV2Manager) GetHistoryTree(
 		return nil, err
 	}
 
-	branches := make([]*persistencespb.HistoryBranch, 0)
-
 	rows, err := m.db.SelectFromHistoryTree(ctx, sqlplugin.HistoryTreeSelectFilter{
 		TreeID:  treeID,
 		ShardID: *request.ShardID,
 	})
 	if err == sql.ErrNoRows || (err == nil && len(rows) == 0) {
 		return &p.GetHistoryTreeResponse{}, nil
 	}
+	branches := make([]*persistencespb.HistoryBranch, 0, len(rows))
```

Comment:
Do we need a max capacity limit here instead of `len(rows)`?

Reply:
`make([]*persistencespb.HistoryBranch, 0, len(rows))` this is allocating mem with max cap of `len(rows)`

the code down below `branches = append(branches, treeInfo.BranchInfo)` will reallocate mem if above limit is reached